Team13/main

HectorBergman opened the pull request:
### Team 13 first stage's simple backend

A simple backend for the first stage of team 13's game.

* Generates houses lined in a row from numbers 1-31 (what is available in the vocabulary as for now
* Displays the correct house in Swedish and English below the houses (The english text will be obscured in the future)
* When a house is clicked, displays correct or incorrect depending on if the house number matched the text
* Houses can increment between eachother by 1 or 2 depending on a random variable
* Amount of houses can be altered by changing houseCount, up to 15 before the game risks breaking
* A "re-randomize" button for getting a new house number
* Simple gameState skeleton not currently in use, intended for going from the main menu to the game to the end screen


### Update 2:

* Added start menu
* Implemented UI for main game and start menu
* Words can be translated and untranslated, except for numbers and streets
* Game can progress, and has end condition (10 points, +1 for correct, -1 for incorrect)
* Back button to quit out of game early

Todo:
- Remove pop-up alerts
- End screen
- Show points screen
- Utilize save.js
- Colorblind-friendly color scheme

HectorBergman committed:
test

HectorBergman committed:
test2

HectorBergman committed:
test push

HectorBergman committed:
it works

HectorBergman committed:
moving around funcs

HectorBergman committed:
Merge branch 'main' into team13/test

SadiaK-art committed:
Add files via upload

Add sandra's images from papunet.net

SadiaK-art committed:
Create house.license

HectorBergman committed:
clean up

HectorBergman committed:
Merge branch 'group13-test' of https://github.com/uu-semp/swedish-learning-app-2025 into team13/test

HectorBergman committed:
oops

HectorBergman committed:
very basic state machine

HectorBergman committed:
re-randomize number

HectorBergman committed:
switching between game states

HectorBergman committed:
small alterations

HectorBergman committed:
github example

LudvigMagdeburg-Official committed:
Added three buttons to main screen

Added three buttons to main screen

HectorBergman committed:
generate houses

HectorBergman committed:
random houses works, added numbers to csv

HectorBergman committed:
correcting leftover debug line

HectorBergman committed:
Pondering how to make buttons work

HectorBergman committed:
Houses display numbers

HectorBergman committed:
Dynamic house buttons

edit houseCount for more or fewer houses

SadiaK-art committed:
feat: add start page (HTML, CSS, JS)

LudvigMagdeburg-Official committed:
Team13/backend/first game (#96)

### Team 13 first stage's simple backend

A simple backend for the first stage of team 13's game.

* Generates houses lined in a row from numbers 1-31 (what is available
in the vocabulary as for now
* Displays the correct house in Swedish and English below the houses
(The english text will be obscured in the future)
* When a house is clicked, displays correct or incorrect depending on if
the house number matched the text
* Houses can increment between eachother by 1 or 2 depending on a random
variable
* Amount of houses can be altered by changing houseCount, up to 15
before the game risks breaking
* A "re-randomize" button for getting a new house number
* Simple gameState skeleton not currently in use, intended for going
from the main menu to the game to the end screen

HectorBergman committed:
Merge branch 'main' into team13/main

LudvigMagdeburg-Official committed:
Merge branch 'main' into team13/frontend/start-page

LudvigMagdeburg-Official committed:
Merge branch 'team13/main' into team13/frontend/start-page

HectorBergman committed:
early stage translation

HectorBergman committed:
Prompt keeps track of translated words

HectorBergman committed:
Simple translation

to do: - and int exception

HectorBergman committed:
swedish number

xFrednet commented on words.csv:
```diff
@@ -1,2 +1,332 @@
-ID,English,Article,Swedish,Literal,Category,Image-url,Audio-url,Team01,Team02,Team03,Team04,Team05,Team06,Team07,Team08,Team09,Team10,Team11,Team13,Team14,Team15,Team16
-use,google sheets,during,development,this,is,for,later,,,,,,,,,,,,,,
+ID,English,Article,Swedish,Swedish_plural,Literal,Category,Image_url,Image_copyright_info,Audio_url,Team01,Team02,Team03,Team04,Team05,Team06,Team07,Team08,Team09,Team10,Team11,Team12,Team13,Team14,Team15,Team16
```
LGTM, it looks like you just downloaded the google sheets file as CSV correct?

xFrednet replied:
The site should always fetch the newest version of the doc 🤔  does it work now?

SadiaK-art committed:
Add files via upload

SadiaK-art committed:
Merge branch 'group13-test' into feature/start-page-ui-sadia

SadiaK-art committed:
added licenses for images

SadiaK-art committed:
deleted extra files sorry

HectorBergman committed:
Update scripts/vocabulary.js

Co-authored-by: Fridtjof Stoldt <xFrednet@gmail.com>

xFrednet reviewed (approved):
My files look good to me :)

HectorBergman committed:
Added streets

HectorBergman committed:
cleaned up index.js

HectorBergman committed:
more cleaning + seperating files

HectorBergman committed:
removed duplicates

HectorBergman committed:
test

SadiaK-art committed:
adjusted header footer to match main menu

SadiaK-art committed:
Update script.js

klintskar committed:
Changes names on some items that have updated names

Some file names were missing a fully descriptive name

HectorBergman committed:
Revert "Changes names on some items that have updated names"

This reverts commit f5dbcae59650a37dd40663a9bf5afe36622d94a7.

HectorBergman committed:
Reapply "Changes names on some items that have updated names"

This reverts commit 52d1099faa63dcd65a5f66b5e30b3693aeae2c80.

HectorBergman committed:
Revert "Reapply "Changes names on some items that have updated names""

This reverts commit 231d61f20281a1d24300507375279991ad2c79e5.

HectorBergman committed:
Reapply "Reapply "Changes names on some items that have updated names""

This reverts commit f8434a03e9ba7e99c584f3d215b48b07c4b1c5b4.

HectorBergman committed:
ran some asset stuff

HectorBergman committed:
gitignore :)

HectorBergman committed:
Team13/backend/translate (#124)

### Backend translation

* Added translatable prompt that can be translated word for word by
clicking them
* Randomized Uppsala-based street names
* Cleaned up and separated js files
* Removed unnecessary text from webpage (mostly debug text)

SadiaK-art committed:
Updated index.html

HectorBergman committed:
Merge branch 'main' into team13/main

LudvigMagdeburg-Official committed:
Merge branch 'feature/start-page-ui-sadia' into team13/frontend/start-page

LudvigMagdeburg-Official committed:
Trying to integrate all the branches with vue this is the skeleton for the html file with vue as a base

LudvigMagdeburg-Official committed:
Small fix to html file.
Have converted the index.js file to use vue by working with the HTML and using the old functions as template everything has been converted to vue. There are still some minor bug which have to be fixed for it to run as intended

LudvigMagdeburg-Official committed:
Everything is working in vue now and I have used the code from the branch with translations and also used the game interface for the game that has houses etc. The only thing here is that the frontpage is not updated with the latest code and the street names which appear are just a list in the index.js file. When a player clicks on a word for translation the player cannot click on the word again for it to go back to the original state

LudvigMagdeburg-Official committed:
I have connected the start page from another branch however did not manage to center it fully

xFrednet reviewed (commented):
Wow, that's a long set of streets. ^^ I think it would make more sense to only store them in your team folder than the main dataset, but this would also be fine

LudvigMagdeburg-Official committed:
Spelet går tillbaka till huvudmenyn när det är klart och boxarna runt order är borttagna så nu är et dotted lines under orden som går att clicka på och dessa ord får upp en shadow runt sig när man hover på dom. Sifferorder är även i bold och understruket och gatunamnet är bara som vanligt med inget speciellt runt

LudvigMagdeburg-Official committed:
Updated function for translations so it now can translate back and forth between swedish and english

SadiaK-art committed:
Added numbers inside houses

SadiaK-art committed:
Add house.png

xFrednet reviewed (approved):
My files LGTM

SadiaK-art committed:
Added on-screen Result Messages

- The alerts have been removed
- Instead, we now display "✖ Wrong house, try again!' or '✔ Correct!" under each house

SadiaK-art committed:
Update Wrong house message to avoid shifting houses

HectorBergman committed:
Revert "Update Wrong house message to avoid shifting houses"

This reverts commit 0d293ad495633413dd7eee38127b9ff729a651d4.

HectorBergman committed:
Revert "Added on-screen Result Messages"

This reverts commit 6ed0773c77794b3732d53576eb821db2127762f3.

HectorBergman committed:
translate-tweak

HectorBergman committed:
Revert "Added numbers inside houses"

This reverts commit e669f7ba5d499b78a52e3a84231e8c49bfd67c29.

HectorBergman committed:
Revert "Add house.png"

This reverts commit 2bcd67238ad2adb001c1d942372b877b39471638.

HectorBergman committed:
Merge branch 'team13/main' into team13/frontend/start-page

HectorBergman committed:
Team13/frontend/start page (#167)

Major update to the Team 13 game.
Full integration between all different frontend, backend and game
logics.
Everything has been integrated and revamped using vue.js
Small bug like game not going back to start page when done and
translations only translating from swedish to english has been fixed.
